File: Javascript/InvertBinaryTree.js
function TreeNode(val,left, right) {
    this.val = (val === undefined) ? 0 : val
    this.left = (left === undefined) ? null : left
    this.right = (right === undefined) ? null : right
}

var invertTree = function(root){

    function swapChild(node){
        if(node === null || (node.left === null & node.right === null)){
            return
        }

        let temp = node.left
        node.left = node.right
        node.right = temp

        swapChild(node.left)
        swapChild(node.right)
    }

    swapChild(root)

    return root
}

File: Python/InvertBinaryTree.py
class TreeNode:

    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right

def invertTree(root):

    def swapChild(node):
        if node is None or (node.left == None and node.right == None):
            return
        
        temp = node.left
        node.left = node.right
        node.right = temp

        swapChild(node.left)
        swapChild(node.right)

    swapChild(root)

    return root